packages/g/gperftools/xmake.lua

```diff
@@ -4,30 +4,86 @@ package("gperftools")
+    add_configs("tcmalloc", {description = "Use tcmalloc", default = true, type = "boolean"})
+    add_configs("profiler", {description = "Use profiler", default = true, type = "boolean"})
     if is_plat("linux") then
         add_configs("unwind", {description = "Enable libunwind support.", default = false, type = "boolean"})
     end
 
     add_deps("cmake")
 
-    on_load("linux", function (package)
-        if package:config("unwind") then
+    if on_check then
+        on_check("windows", "mingw", function (package)
+            assert(package:config("minimal"), "package(gperftools): only tcmalloc_minimal is supported on Windows or MinGW")
+            if is_plat("windows") then
+                assert(package:version():ge("2.16") or package:version():eq("2.10"), "package(gperftools): requires version = 2.10 or >= 2.16 for Windows")
+            else
+                assert(package:version():ge("2.16"), "package(gperftools): requires version >= 2.16 for MinGW")
+            end
+        end)
+        on_check("macosx", function (package)
+            if not package:version():ge("2.14") then
+                if not (package:version():eq("2.10") and macos.version():le("12")) then
+                    raise("package(gperftools): requires version >= 2.14 for macOS")
+                end
+            end
+        end)
+    end
+
+    on_load(function (package)
+        if package:is_plat("linux") and package:config("unwind") then
             package:add("deps", "libunwind")
         end
+
+        if package:config("tcmalloc") then
+            local libsuffix = package:config("minimal") and "_minimal" or ""
+            libsuffix = package:is_debug() and libsuffix .. "_debug" or libsuffix
+            package:add("links", "tcmalloc" .. libsuffix)
+        end
+
+        if package:config("profiler") then
+            package:add("links", "profiler")
+        end
     end)
 
-    on_install("windows", "macosx", "linux", function (package)
+    on_install("windows|!arm64", "macosx", "linux", "mingw", function (package)
         local configs = {"-DBUILD_TESTING=OFF", "-Dgperftools_build_benchmark=OFF"}
-        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:is_debug() and "Debug" or "Release"))
         table.insert(configs, "-DGPERFTOOLS_BUILD_STATIC=" .. (package:config("shared") and "OFF" or "ON"))
+        table.insert(configs, "-Dgperftools_build_minimal=" .. (package:config("minimal") and "ON" or "OFF"))
         if package:is_plat("linux") then
             table.insert(configs, "-Dgperftools_enable_libunwind=" .. (package:config("unwind") and "ON" or "OFF"))
         end
-        import("package.tools.cmake").install(package, configs)
+
+        io.replace("CMakeLists.txt", "if(MINGW OR MSVC)", "if(MINGW OR MSVC)\nlink_libraries(psapi synchronization shlwapi)", {plain = true})
```

Comment:
On Windows and MinGW platforms, tcmalloc requires linking to specific system libraries. However, `link_libraries()` in the `CMakeLists.txt` is applied too late in the build process, causing linking issues. See https://github.com/gperftools/gperftools/blob/e1014dead2029b341d06027b4f2b5562d799d5b1/CMakeLists.txt#L451-L467

A quick solution can be implemented by leveraging another `if(MINGW OR MSVC) ... endif()` block at the beginning of the `CMakeLists.txt` file:
https://github.com/gperftools/gperftools/blob/e1014dead2029b341d06027b4f2b5562d799d5b1/CMakeLists.txt#L32C1-L37C8